Extensions/ArtifactEngine/Providers/filesystemProvider.ts

```diff
@@ -33,21 +33,26 @@ export class FilesystemProvider implements models.IArtifactProvider {
         return promise;
     }
 
-    public putArtifactItem(item: models.ArtifactItem, stream: Stream.Readable): Promise<models.ArtifactItem> {
+    public putArtifactItem(item: models.ArtifactItem, stream: NodeJS.ReadableStream): Promise<models.ArtifactItem> {
         return new Promise(async (resolve, reject) => {
             const outputFilename = path.join(this._rootLocation, item.path);
 
             // create parent folder if it has not already been created
             const folder = path.dirname(outputFilename);
             this.ensureDirectoryExistence(folder);
 
-            Logger.logMessage('Downloading '+ item.path + ' to '+ outputFilename);
+            Logger.logMessage('Downloading ' + item.path + ' to ' + outputFilename);
             const outputStream = fs.createWriteStream(outputFilename);
             stream.pipe(outputStream);
             stream.on("end",
                 () => {
                     Logger.logMessage(`Downloaded '${item.path}' to '${outputFilename}'`);
+                    if (!item.metadata) {
```

Comment:
when will this happen?

Reply:
depened on the template currently all templates have metadata, if there is no metadata we should explicitly reject here.